### PR TITLE
Some clean ups and add support for Fmx grid viewmodels

### PR DESCRIPTION
**Pull request by mirko-bianco**

### CODE REVIEW

File: FidoGui.md [hunk 1]
```diff
@@ -23,3 +23,10 @@ The `MVVMExample` project shows how binding can be achieved, the main points are
     Guibinding.MethodsSetup<ISongViewModel, TSongView>(FSongViewModel, Self);
   ```
 
+The `GridViewModelExample` project shows how binding of a grid and a list can be achieved, the main points are:
+
+- Make use of  the `TDelegateObservable` as base for the entity that needs binding. 
+- Make use of  the `TAbstractReadonlyGridView<T>`or `TAbstractWriteableGridView<T>` as base for the grid view model that will consume the List.
+- Make use of  the `IFilteredGridViewModelHelper<T, FilterPayload>` helper if you need to filter the list through the view model.
+- Make use of  the `ISortedByColumnGridViewModelHelper<TT>` helper if you need to sort the list through the grid columns position and header click.
```
Comment (BasvRossem) [inline FidoGui.md]:
```suggestion
- Make use of the `TDelegateObservable` as base for the entity that needs binding. 
- Make use of the `TAbstractReadonlyGridView<T>`or `TAbstractWriteableGridView<T>` as base for the grid view model that will consume the List.
- Make use of the `IFilteredGridViewModelHelper<T, FilterPayload>` helper if you need to filter the list through the view model.
- Make use of the `ISortedByColumnGridViewModelHelper<T>` helper if you need to sort the list through the grid columns position and header click.
```
Remove the extra spaces, they aren't rendered, but just in case

File: source/Collections/Fido.Collections.DeepObservableList.pas [hunk 2]
```diff
@@ -75,30 +106,160 @@ procedure TDeepObservableList<T>.Broadcast(const Description: string);
...
+var
+  Observable: IObservable;
+begin
+  inherited;
+
+  for var Item in values do
+    if Supports(Item, IObservable, Observable) then
+       Observable.RegisterObserver(Self);
+
+  FDelegatedObservable.Broadcast('Items added');
```
Comment (BasvRossem) [inline source/Collections/Fido.Collections.DeepObservableList.pas]: Shouldn't you use string constants here, so the user does not have to take spelling into account when listening to these messages?

File: source/Fido.Functional.Tries.pas [hunk 3]
```diff
@@ -105,10 +105,17 @@ function &Try<T>.Map<TOut>(const Func: Context<T>.FunctorFunc<TOut>): TryOut<TOu
...
 begin
   LValue := FValue;
-  Result := TryOut<TOut>.New(function: TOut
+//  Result := TryOut<TOut>.New(function: Context<TOut>
+//    begin
+//      Result := Context<TOut>.New(function: TOut
+//        begin
+//          Result := LValue.Map<TOut>(Func);
+//        end);
+//    end);
```
Comment (BasvRossem) [inline source/Fido.Functional.Tries.pas]: Can these be removed?

File: source/Web/Client/WebSocket/Fido.Web.Client.WebSocket.TcpClient.pas [hunk 4]
```diff
@@ -31,6 +31,7 @@ interface
   System.SyncObjs,
   System.Math,
   System.DateUtils,
+  System.Net.HttpClientComponent,
```
Comment (BasvRossem) [inline source/Web/Client/WebSocket/Fido.Web.Client.WebSocket.TcpClient.pas]: Why is this needed?

File: source/Collections/Fido.Collections.DeepObservableList.pas
Comment (BasvRossem) [inline source/Collections/Fido.Collections.DeepObservableList.pas:1]: Are there tests for this file?

Comment (mirko-bianco) [reply source/Collections/Fido.Collections.DeepObservableList.pas:1]: nope...

**Review by BasvRossem — APPROVED** [review]
